### PR TITLE
Opencensus bridge: migrate from README to Go docs

### PR DESCRIPTION
**Pull request by dashpole**
I was inspired by what I learned from @pellared.

This migrates from a README to godocs, and adds an example_test.go

It removes some content that probably isn't relevant for users about making context propagation work between the libraries.

It adds a section on the limitations of the metric bridge.

### CODE REVIEW

File: bridge/opencensus/example_test.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	// rather than OpenCensus.
+	octrace.DefaultTracer = opencensus.NewTracer(tracer)
+}
+
+func ExampleNewMetricProducer() {
+	// Create the OpenCensus Metric bridge.
+	bridge := opencensus.NewMetricProducer()
+	// Add the bridge to your reader using metric.WithProducer().
+	// If using a push exporter, such as OTLP, use metric.NewPeriodicReader.
+	// If using the prometheus exporter, use prometheus.WithProducer.
```
Comment (pellared) [inline bridge/opencensus/example_test.go:38]: I think something is wrong here (at least for me it is hard to follow). How about something like

```suggestion
	// Add the bridge as a producer to your reader.
	// If using a push exporter, such as OTLP exporter,
	// use metric.NewPeriodicReader with metric.WithProducer option.
	// If using a pull exporter which acts as a reader, such as prometheus exporter,
	// use a dedicated option like prometheus.WithProducer.
```

**Review by MrAlias — APPROVED** [review]

**Review by pellared — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]
